### PR TITLE
[SOL-77] Add basic relayer history UI

### PR DESCRIPTION
**Pull request by gitteri**
image

image

### CODE REVIEW

File: src/app/api/v1/transfer/[id]/route.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+'use server';
+
+import { NextRequest, NextResponse } from 'next/server';
+import { validate } from 'uuid';
+import { SplTransfer, TransactionStatus } from '@/app/types/splTransfer';
+import { getSplTransfer } from '@/logic/transactionEngine';
+
+// This type is used to return the transaction to the client.
+// It is used to hide certain fields from the public.
+export type PublicSplTransfer = Omit<SplTransfer, 'signedTransactionBytes' | 'requestedByIp'> & {
```
Comment (kilogold) [inline src/app/api/v1/transfer/[id]/route.ts:10]: Why hide `signedTransactionBytes`?
This is going to be public on the ledger.

File: src/app/config/mint.ts [hunk 2]
```diff
@@ -9,6 +9,10 @@ const supportedMints: Record<string, MintInfo> = {
     address: '4zMMC9srt5Ri5X14GAgXhaHii3GnPAEERYPJgZJDncDU', // devnet
     decimals: 6,
   },
+  'USDT': {
+    address: '4zMMC9srt5Ri5X14GAgXhaHii3GnPAEERYPJgZJDncDU', // lol what's the devnet USDT address?? can't find it anywhere.
+    decimals: 6,
+  },
```
Comment (kilogold) [inline src/app/config/mint.ts]: Let's use PYUSD!
```suggestion
  'PYUSD': {
    address: 'CXk2AMBfi3TwaEL2468s6zP8xq9NxTXjp9gjMgzeUynM',
    decimals: 6,
  },
```

File: src/logic/transactionEngine.ts [hunk 3]
```diff
@@ -1,18 +1,53 @@
...
+  return createAssociatedTokenAccountInstruction(feePayerPublicKey, newAccount, ownerPublicKey, mintPublicKey)
+}
+
 export async function createSplTransfer(sender: string, destination: string, amount: string, mintSymbol: string): Promise<SplTransfer> {
   const mint = getMintInfo(mintSymbol);
   const relayWallet = EmbeddedWallet.get();
 
-  const RELAY_FEE = '500000'; // 0.50 USDC/USDT (6 decimal places)
+  // TODO: make this dynamic and based on if the token account needs to be created
+  const RELAY_FEE = '50000'; // 0.05 USDC/USDT (6 decimal places)
```
Comment (kilogold) [inline src/logic/transactionEngine.ts]:
```suggestion
  const RELAY_FEE = '50000'; // 0.05 USDC/PYUSD (6 decimal places)
```

File: src/logic/transactionEngine.ts [hunk 4]
```diff
@@ -22,32 +57,41 @@ export async function createSplTransfer(sender: string, destination: string, amo
     mint.address
   );
 
+  const amountLamports = mint.decimals === 0 ? amount : new Decimal(amount).mul(new Decimal(10).pow(mint.decimals)).toString();
```
Comment (kilogold) [inline src/logic/transactionEngine.ts]: Maybe we can use [amountToUiAmount ](https://solana-labs.github.io/solana-program-library/token/js/functions/amountToUiAmount.html) here. Does the same, adhering to conventions.

File: src/logic/transferTracker.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+ * Updates the transfer database record with details from a Helius enriched transaction.
+ * @param {EnrichedTransaction} enrichedTransaction - The enriched transaction from Helius.
+ * @returns {Promise<void>}
+ */
+export async function updateTransferDetails(enrichedTransaction: EnrichedTransaction): Promise<void> {
+  try {
+    const { signature, instructions, slot, timestamp, fee } = enrichedTransaction;
+
+    const memoIx = instructions.find((instruction) => {
+      return instruction.programId.toString() === 'MemoSq4gqABAXKb96qnH8TysNcWxMyWCqXgDLGmfcHr';
```
Comment (kilogold) [inline src/logic/transferTracker.ts]: Noticed this address in a few places. Consider making a const.
Unlikely to change, but Token/Token22 sets precedent.

**Review by kilogold — CHANGES_REQUESTED** [review]

File: src/app/api/v1/transfer/[id]/route.ts (hunk 1 above)
Comment (gitteri) [reply src/app/api/v1/transfer/[id]/route.ts:10]: honestly thinking about just removing it.. not sure what that gives us

File: src/app/config/mint.ts (hunk 2 above)
Comment (gitteri) [reply src/app/config/mint.ts]: agreed let's do it in a follow up when we enable token 2022

File: src/logic/transactionEngine.ts (hunk 4 above)
Comment (gitteri) [reply src/logic/transactionEngine.ts]: looks like this does a network call to verify decimals and then do the conversion. is that worth the extra network call?

Comment (kilogold) [reply src/logic/transactionEngine.ts]: The SDK can be improved in the backend. Using the SDK guarantees compliance & best practices.
If everyone rolled their own math, we're prone to inconsistencies across the ecosystem.

**Review by kilogold — APPROVED** [review]
I can work with this. Ship it.

**Pull request merged**